code/modules/reagents/chemistry/reagents/medicine.dm

```diff
@@ -1001,19 +1001,14 @@
 	reagent_state = LIQUID
 	color = "#FFDCFF"
 	taste_description = "stability"
+	var/list/drug_list = list("crank","methamphetamine","space_drugs","psilocybin","ephedrine","epinephrine","stimulants","bath_salts","lsd","thc")
 
 /datum/reagent/medicine/haloperidol/on_mob_life(mob/living/M)
 	var/update_flags = STATUS_UPDATE_NONE
-	M.reagents.remove_reagent("crank", 5)
-	M.reagents.remove_reagent("methamphetamine", 5)
-	M.reagents.remove_reagent("space_drugs", 5)
-	M.reagents.remove_reagent("psilocybin", 5)
-	M.reagents.remove_reagent("ephedrine", 5)
-	M.reagents.remove_reagent("epinephrine", 5)
-	M.reagents.remove_reagent("stimulants", 3)
-	M.reagents.remove_reagent("bath_salts", 5)
-	M.reagents.remove_reagent("lsd", 5)
-	M.reagents.remove_reagent("thc", 5)
+	for(var/I in M.reagents.reagent_list)
+		var/datum/reagent/R = I
+		if(drug_list.Find(R.id))
```

Comment:
technically you don't even need this `.Find`, for it will be handled in `.remove_reagent` on the next line

Reply:
.. that is, if you iterate over items in `drug_list` and not in `M.reagents.reagent_list` (which I assumed you do, but didn't read carefully enough).